openedx/core/lib/block_structure/transformer.py

```diff
@@ -2,14 +2,40 @@
 This module provides the abstract base class for all Block Structure
 Transformers.
 """
-from abc import abstractmethod
+
+
+class BlockStructureTransformerMetaclass(type):
+    """
+    A metaclass which verifies subclasses of BlockStructureTransformer.
+    """
+    def __new__(mcs, name, bases, attrs):
+        new_class = super(BlockStructureTransformerMetaclass, mcs).__new__(mcs, name, bases, attrs)
+
+        def _is_method_overridden(method_name):
+            """
+            Returns whether a method of the given method_name is overriden.
+            """
+            return getattr(new_class, method_name) != getattr(BlockStructureTransformer, method_name)
+
+        if name != 'BlockStructureTransformer':
+            block_filter_overriden = _is_method_overridden('transform_block_filter')
+            transform_overriden = _is_method_overridden('transform')
```

Comment:
Doh!  Why didn't I think of that?  I think because I was initially thinking a Transformer could support both types of transformation methods - and we would use either in different situations.  But the code evolved since then.  But yeah - much better than creating a Meta class to enforce exclusive-or methods.